listener.go

```diff
@@ -186,10 +188,24 @@ func (l *listener) doOnConn(conn net.Conn) error {
 		initMsg           = make([]byte, clientInitSize)
 	)
 
+	// Handle pausing of idletiming to keep it from superseding our InitMsgTimeout
+	unpauseIdleTiming := func() {
+		// do nothing
+	}
+	netx.WalkWrapped(conn, func(wrapped net.Conn) bool {
+		idleConn, ok := wrapped.(*idletiming.IdleTimingConn)
+		if ok {
+			unpauseIdleTiming = idleConn.Pause()
+			return false
+		}
+		return true
+	})
```

Comment:
To be honest, I don't love the coupling of the callee anticipating how it might be called (lampshade knowing that it may get invoked using idletiming).  Seems fairly harmless though and I don't see an easy way to avoid it.

Reply:
Yeah, I don't love it either.